platform/engineering/design-docs/2020-05-26-content-validation.md

```diff
@@ -0,0 +1,395 @@
+3. Run the content validation script
+4. Confirm that the script reports the newly introduced broken link
+
+##### Accessibility errors
+
+###### Positive
+
+1. Confirm that the content build has zero accessibility errors
+2. Run the content validation script
+3. Confirm that the script reports zero accessibility errors
+
+###### Negative
+
+1. Confirm that the content build has zero accessibility errors
+2. Modify an HTML file, like [`header.html`](https://github.com/department-of-veterans-affairs/content-build/blob/master/src/site/includes/header.html), to have an accessibility error
+3. Confirm that the script reports the newly introduced accessibility error
+
+#### CI
+
+##### Positive
+
+1. Confirm that the content build has zero accessibility errors and zero broken links
+2. Run the job that calls the content validation script
+3. Confirm that a report is sent to the #cms-team Slack channel
+4. Confirm that the report includes zero accessibility errors and zero broken links
+
+##### Negative
+
+1. Confirm that the content build has zero accessibility errors and zero broken links
+2. Add a broken link, and an accessibility error
+3. Run the job that calls the content validation script
+4. Confirm that a report is sent to the #cms-team Slack channel
+5. Confirm that the report includes the newly introduced accessibility error
+6. Confirm that the report includes the newly introduced broken link
+7. Fix the newly introduced accessibility error
+8. Fix the newly introduced broken link
+9. Run the job that calls the content validation script
+10. Confirm that the report includes zero accessibility errors and zero broken links
+
+### Logging
+
+Currently, invalid content is logged in the build log during staging deploys. And failed builds trigger generic [Slack notifications](https://dsva.slack.com/archives/C37M86Y8G/p1591099447374800).
+
+The updated content validation approach will continue logging invalid content errors in the build log, and will also post an automated report of the invalid content in the #cms-team Slack channel.
+
+### Debugging
+
+- Local build logs
+- Jenkins build logs
+- CircleCI build logs
+- Slack notifications
+
+### Caveats
+
+### Security Concerns
+
+There are no new security concerns with a separated content validation process.
+
+### Privacy Concerns
+
+There are no new privacy concerns with a separated content validation process.
+
+### Open Questions and Risks
+
+#### Risks
+
+- Neither the current content validation nor the proposed content validation will **prevent** invalid content from making it to the live VA.gov site. But this proposal would prevent unrelated teams from being blocked by invalid content.
+
+#### Questions
+
+- What is the optimal frequency to run the scheduled job? Daily seems too slow. Every 10 minutes might be too fast for our builds. Hourly seems feasible. Is hourly acceptable? Should we be c concern about noise from a higher frequency?
```

Comment:
I think hourly is OK, but errors should pile up. Like, if one error reports and isn't fixed it should be reported again after an hour and keep reporting until it is fixed

Reply:
I'd suggest that this should run with every content release.  Otherwise the reporting could be completely out of sync with what is actually out there and have editors thrashing to chase down reported issues they already resolved.

Reply:
Yeah, I agree there, but I would also scan hourly in case there's a large gap in between content releases. So something doesn't sit there for too long and risk getting missed.

Reply:
The hourly idea came from the previous platform lead:

>how long do we anticipate this job taking? could we run it more frequently? (hourly? every 10 minutes?) - https://github.com/department-of-veterans-affairs/va.gov-team/pull/9658#discussion_r458442981